### ISSUE TITLE
some of the Shape extension's function are not available, e.g. `XShapeCombineRegion` 

### ISSUE THREAD

**Issue by makemeunsee** [psychon/x11rb#667]
Hi,

I'm trying to implement an input passthrough behavior similar to this:

https://github.com/glfw/glfw/pull/1568/files#diff-8c3a21538cd2ef071ab75ebf6ff2693fd14d4167d7096f6d41fd8ce2a0fe54e4R2633-R2658

and got stuck in particular because `XShapeCombineRegion` has currently no binding.

I may try to attempt a fix myself, although I would welcome a workaround too.

**Comment by notgull** [comment]
[A binding already exists in the `shape` extension.](https://docs.rs/x11rb/latest/x11rb/protocol/shape/fn.combine.html)

**Comment by makemeunsee** [comment]
> [A binding already exists in the `shape` extension.](https://docs.rs/x11rb/latest/x11rb/protocol/shape/fn.combine.html)

This one has a slightly different signature, combining 2 windows, whereas `XShapeCombineRegion` combines a region and a window. Or regions and windows are interchangeable in practice?

Anyway the same effect (as far as I can tell) can be achieved with:
```rust
    let rw = RegionWrapper::create_region(&conn, &[]).unwrap();

    let set_shape_request = SetWindowShapeRegionRequest {
        dest: win_id,
        dest_kind: shape::SK::BOUNDING,
        x_offset: 0,
        y_offset: 0,
        region: 0,
    };
    let _ = set_shape_request.send(&conn);

    let set_shape_request = SetWindowShapeRegionRequest {
        dest: win_id,
        dest_kind: shape::SK::INPUT,
        x_offset: 0,
        y_offset: 0,
        region: rw.region(),
    };
    let _ = set_shape_request.send(&conn);
    let _ = destroy_region(&conn, rw.region());
```

so we might as well close the issue.

**Comment by psychon** [comment]
I tried to find the source code for `XShapeCombineRegion`, because sometimes those Xlib libraries "do something" that does not map to the protocol that easily. Via https://codesearch.debian.net/search?q=XShapeCombineRegion+package%3A%5CQlib&literal=1 I found the right file. I would link to, but that's not a permalink, so instead here is a permalink into the "proper source":

https://github.com/freedesktop/libXext/blob/490a25e6f8a4d2482af4364c700b68ad11a4d10b/src/XShape.c#L188-L231

`XShapeCombineRegion` seems to send a `ShapeRectangles` request (`X_ShapeRectangles`) with `ordering = YXBanded`. Besides that, I cannot see anything special.

And no, I do not know why the protocol request and the lib function have so different names.
